### DISCUSSION TITLE
jump to not php file

### DISCUSSION THREAD

**Discussion by zordsdavini** [phpactor/phpactor#2387] [Ideas]
In Symfony projects there are yaml or twig files. The idea is to add jumping to these files:
- jump to twig file from php
- jump to routing files if controller action has such route

Can be related to `symfony.enabled` config.

**Comment by mamazu** [comment]
They are both great ideas. Here are my two cents for both of them.

The first one sounds interesting and would be quite easy to implement if it weren't for custom template namespaces like `@SomeBundle/master.html.twig`.

The second one is going go to be a challenge if the route definition isn't also inside a php file as phpactor right now doesn't parse any config files. In the projects I work with I have moved all configuration to php and that has made life much easier but I get that this isn't going to be a good fit for every project.

**Comment by zordsdavini** [comment]
Additionally to this '@SomeBundle/xxx' anotations could be indexed and allow to jump, ex., validators

**Comment by dantleech** [comment]
You can use the `var/cache/url_matching_routes.php` file to map route names to controllers and vice versa (we already use the container dump XML file for service metadata).  Jumping _to_ routing files is likely to me more of a pain.
